src/Benchmarks/Program.cs

```diff
@@ -84,6 +85,11 @@ public static void Main(string[] args)
                         }
 #endif
                     })
+                    .ConfigureHttpJsonOptions(jsonOptions =>
+                    {
+                        jsonOptions.SerializerOptions.Encoder = null;
+                        jsonOptions.SerializerOptions.AddContext<Middleware.CustomJsonContext>();
```

Comment:
1. AddContext is going to be obsoleted as per https://github.com/dotnet/runtime/issues/83280.
2. I'm a bit confused at how this works today. `AddContext` [will put the `CustomJsonContext` at the end of the TypeInfoResolver](https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L175-L179), after the "default" resolver. Doesn't this mean the reflection based type info resolver still gets used over the `CustomJsonContext`?

cc @eiriktsarpalis

Reply:
This worked temporarily until AddContext was fixed to properly combine contexts. I think my local branch fixed that but never pushed it.

Reply:
You should use this to ensure equivalent behavior with the .NET 8 build
```suggestion
                        jsonOptions.SerializerOptions.TypeInfoResolver = JsonTypeInfoResolver.Combine(jsonOptions.SerializerOptions.TypeInfoResolver, Middleware.CustomJsonContext.Default);
```

Reply:
Shouldn't it be `Combine(Middleware.CustomJsonContext.Default, jsonOptions.SerializerOptions.TypeInfoResolver);`? Or does order not matter with `Combine`?

Reply:
You're right, I messed up the order in my suggestion.